bin/classify_SingleR.R

```diff
@@ -55,19 +56,28 @@ if (!file.exists(opt$sce_file)) {
   stop("Missing SCE file")
 }
 
-# check that output files have the righr extensions
+# check that output files have the right extensions
 if (!(stringr::str_ends(opt$output_singler_results_file, ".rds"))) {
   stop("output SingleR result file name must end in .rds")
 }
 if (!(stringr::str_ends(opt$output_singler_annotations_file, ".tsv"))) {
   stop("output SingleR annotations file name must end in .tsv")
 }
 
-# check that references all exist
+# check that reference exists and filename is properly formatted
 singler_model_file <- opt$singler_model_file
 if (!file.exists(singler_model_file)) {
   stop(glue::glue("Provided model file {singler_model_file} is missing."))
 }
+if (!(stringr::str_ends(singler_model_file, "_model.rds"))) {
+  stop(glue::glue("Provided model file {singler_model_file} must end in .rds."))
+}
+
+# get & check reference name
+reference_name <- stringr::str_match(singler_model_file, "(\\w+)_model.rds")[2]
+if (reference_name == "") {
+  stop(glue::glue("Provided model file name must be formatted as `<model_name>_model.rds`"))
+}
```

Comment:
I think the `str_remove` is more clear and simpler.

Your already checked above that the file had the right ending, so the second check here will only be triggered if the full file name was "_model.rds". 
```suggestion
reference_name <- stringr::str_remove(singler_model_file, "_model.rds$")
if (reference_name == "") {
  stop(glue::glue("Provided model file name must be formatted as `<model_name>_model.rds`"))
}
```